spec/std/big_rational_spec.cr

```diff
@@ -0,0 +1,148 @@
+require "spec"
+require "big_rational"
+
+def br(n, d)
+  BigRational.new(n, d)
+end
+
+def test_comp(val, less, equal, greater, file = __FILE__, line = __LINE__)
+  (val < greater).should eq(true), file, line
+  (greater < val).should eq(false), file, line
+  (val <=> greater).should eq(-1), file, line
+  (greater <=> val).should eq(1), file, line
+
+  (val == equal).should eq(true), file, line
+  (equal == val).should eq(true), file, line
+  (val <=> equal).should eq(0), file, line
+  (equal <=> val).should eq(0), file, line
+
+  (val > less).should eq(true), file, line
+  (less > val).should eq(false), file, line
+  (val <=> less).should eq(1), file, line
+  (less <=> val).should eq(-1), file, line
+end
+
+describe BigRational do
+  it "initialize" do
+    BigRational.new(BigInt.new(10), BigInt.new(3))
+               .should eq(BigRational.new(10, 3))
+
+    expect_raises(DivisionByZero) do
+      BigRational.new(BigInt.new(2), BigInt.new(0))
+    end
+
+    expect_raises(DivisionByZero) do
+      BigRational.new(2, 0)
+    end
+  end
+
+  it "#numerator" do
+    br(10, 3).numerator.should eq(BigInt.new(10))
+  end
+
+  it "#denominator" do
+    br(10, 3).denominator.should eq(BigInt.new(3))
+  end
+
+  it "#to_s" do
+    br(10, 3).to_s.should eq("10/3")
+    br(90, 3).to_s.should eq("30")
+    br(1, 98).to_s.should eq("1/98")
+
+    r = BigRational.new(8243243, 562828882)
+    r.to_s(16).should eq("7dc82b/218c1652")
+    r.to_s(36).should eq("4woiz/9b3djm")
+  end
+
+  it "#to_f64" do
+    r = br(10, 3)
+    f = 10.to_f64 / 3.to_f64
+    r.to_f64.should be_close(f, 0.001)
+  end
+
+  it "#to_f" do
+    r = br(10, 3)
+    f = 10.to_f64 / 3.to_f64
+    r.to_f.should be_close(f, 0.001)
+  end
+
+  it "#to_f" do
```

Comment:
This frequently happens to me too (copy past test, forget to change the name). I wonder if it would be a good idea to raise an error if this happens so it's easier to spot.